rq/scheduler.py

```diff
@@ -161,8 +161,12 @@ def enqueue_scheduled_jobs(self):
                         try:
                             queue._enqueue_job(job, pipeline=pipeline, at_front=bool(job.enqueue_at_front))
                         except DeserializationError:
-                            self.log.error('Deserialization error while trying to enqueueing job %d', job.id, exc_info=True, stack_info=True)
-                        registry.remove(job, pipeline=pipeline)
+                            self.log.error('RQ Scheduler Error: deserialization error while trying to enqueueing job %d', job.id, exc_info=True, stack_info=True)
+                        except Exception as e:
+                            self.log.error('RQ Scheduler Error: while trying to enqueue job %d: %s', job.id, str(e), exc_info=True, stack_info=True)
+                        finally:
+                            # Always remove the job from registry, even if there was an error
+                            registry.remove(job, pipeline=pipeline)
```

Comment:
❓ This means if there's a network error the job is lost and won't be retried?
Do we have not have a failed queue we can put this on?

Not familiar with the mechanism on how things end up in the failed queue, I can look!

Reply:
I meant, this line:
> queue._enqueue_job

I assumed that it meant the scheduler process calls the Redis store and there are network calls involved. Is it not the case? (my mental model might be totally off!).

> So it means that if the code fails to schedule the job, it will issue a log error, and won't retry to reschedule the job. However the job will still be there in Redis.

Ah that makes sense!